indico/modules/categories/templates/display/event_list.html

```diff
@@ -0,0 +1,158 @@
+                             before=none, after=none, message_first=true) %}
+    {% set message=caller() %}
+    {% macro render_message() %}
+        <div class="topBar">
+            <div class="content smaller">
+                <span class="js-show-message">
+                    {{ message }}
+                    <a class="js-toggle-list" href="#">{% trans %}Show{% endtrans %}</a>
+                </span>
+                <span class="js-hide-message">
+                    {{ message }}
+                    <a class="js-toggle-list" href="#">{% trans %}Hide{% endtrans %}</a>
+                </span>
+            </div>
+        </div>
+    {% endmacro %}
+
+    {% macro render_list() %}
+        <div class="events"
+             data-event-list-url="{{ url_for('.event_list', category) }}"
+             data-event-list-after="{{ after }}"
+             data-event-list-before="{{ before }}">
+            {%- if events_by_month -%}
+                {{ event_list_block(events_by_month, format_event_date, is_recent, happening_now) }}
+            {%- endif -%}
+        </div>
+    {% endmacro %}
+
+    <div class="{{ class }}">
+        {% if message_first %}
+            {{ render_message() }}
+            {{ render_list() }}
+        {% else %}
+            {{ render_list() }}
+            {{ render_message() }}
+        {% endif %}
+    </div>
+{% endmacro %}
+
+{% macro event_list(events_by_month, format_event_date, is_recent, happening_now, category, future_event_count,
+                    past_event_count, future_threshold, past_threshold, future_events_by_month=[],
+                    past_events_by_month=[]) %}
+    <div class="event-list">
+        {% if future_event_count %}
+            {% call hidden_events_block('future-events', category, future_events_by_month, format_event_date,
+                                        is_recent, happening_now, after=future_threshold) %}
+                {% trans count=future_event_count -%}
+                    There is {{ count }} event in the future.
+                {%- pluralize -%}
+                    There are {{ count }} events in the future.
+                {%- endtrans %}
+            {% endcall %}
+        {% endif %}
+        <div>
+            {{ event_list_block(events_by_month, format_event_date, is_recent, happening_now) }}
+        </div>
+        {% if past_event_count %}
+            {% call hidden_events_block('past-events', category, past_events_by_month, format_event_date, is_recent,
+                                        happening_now, before=past_threshold, message_first=false) %}
+                {% trans count=past_event_count -%}
+                    There is {{ count }} event in the past.
+                {%- pluralize -%}
+                    There are {{ count }} events in the past.
+                {%- endtrans %}
+            {% endcall %}
+        {% endif %}
+    </div>
+    <script type="text/javascript">
```

Comment:
type is not needed in HTML5, but wrap it in a (strict-mode) function to contain all those globals